lib/memory_profiler/reporter.rb

```diff
@@ -113,11 +113,17 @@ def object_list(generation)
           # we do memsize first to avoid freezing as a side effect and shifting
           # storage to the new frozen string, this happens on @hash[s] in lookup_string
           memsize = ObjectSpace.memsize_of(obj)
-          string = klass == String ? helper.lookup_string(obj) : nil
+          string = if klass == String
+            shared = ObjectSpace.dump(obj)["shared"]
```

Comment:
this is going to add a log of cost to the report ... we are certainly going to need to make this a switch of sorts.

Reply:
dump returns a string ... if for some reason the json has shared: false in it this would still be true

Reply:
> this is going to add cost to the report ... 

What do you mean by that? because the call to `.dump` will generate a string that would otherwise be iterated on? If so, isn't `.lookup_string` doing the same?

> dump returns a string...

You're right, still need to parse it. Or would you suggest alternative ways of looking about this? I could also easily use memsize to guard this operation, i.e. skip if memsize is not 40.